src/main/catalog/Catalog.ts

```diff
@@ -162,8 +162,18 @@ export class Catalog {
                 if (res.length > 0) {
                     // Pick a random service from the list of healthy services
                     const ID = Math.floor(Math.random() * res.length)
-                    const address: string = res[ID].Service.Address
-                    const port: number = res[ID].Service.Port || 80
+                    /*
+                    https://www.consul.io/api/catalog.html#serviceaddress
+                    ServiceAddress is the IP address of the service host — if empty, node
+                    address should be used
+                    */
+                    const pickedNode = res[ID]
+                    const address: string =
+                        pickedNode.Service.Address &&
+                        pickedNode.Service.Address !== ''
+                            ? pickedNode.Service.Address
+                            : pickedNode.Node.Address
+                    const port: number = pickedNode.Service.Port || 80
```

Comment:
Not sure if this could happen, but could pickedNode.Node.Address be empty also? If so, should there be an error thrown?

Reply:
Good question. However from the documentation in consul it is not possible. Registering a catalog entry has 'Address' as a required field. https://www.consul.io/api/catalog.html#parameters